src/modules/ModuleRunner.ts

```diff
@@ -17,28 +17,118 @@ limitations under the License.
 import { safeSet } from "matrix-js-sdk/src/utils";
 import { TranslationStringsObject } from "@matrix-org/react-sdk-module-api/lib/types/translations";
 import { AnyLifecycle } from "@matrix-org/react-sdk-module-api/lib/lifecycles/types";
-import { AllExtensions } from "@matrix-org/react-sdk-module-api/lib/types/extensions";
-import { DefaultCryptoSetupExtensions } from "@matrix-org/react-sdk-module-api/lib/lifecycles/CryptoSetupExtensions";
-import { DefaultExperimentalExtensions } from "@matrix-org/react-sdk-module-api/lib/lifecycles/ExperimentalExtensions";
-import { RuntimeModule } from "@matrix-org/react-sdk-module-api";
+import {
+    DefaultCryptoSetupExtensions,
+    ProvideCryptoSetupExtensions,
+} from "@matrix-org/react-sdk-module-api/lib/lifecycles/CryptoSetupExtensions";
+import {
+    DefaultExperimentalExtensions,
+    ProvideExperimentalExtensions,
+} from "@matrix-org/react-sdk-module-api/lib/lifecycles/ExperimentalExtensions";
 
 import { AppModule } from "./AppModule";
 import { ModuleFactory } from "./ModuleFactory";
 
 import "./ModuleComponents";
 
+class ExtensionImplementationMap {
+    public hasDefaultCryptoSetupExtension: boolean = true;
+    public hasDefaultExperimentalExtension: boolean = true;
+}
+
+/**
+ * Handles and manages any extensions provided by modules
+ */
+class ExtensionsManager {
```

Comment:
In this PR the class is not exported as it is used only in ModuleRunner. 
The plan is to move this class to matrix-react-sdk-module-api in a separate PR, at which point it will be exported. So we should avoid anything taking dependence on ExtensionsManager (other than ModuleRunner).